prov/efa/src/efa_domain.c

```diff
@@ -316,8 +327,8 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 		}
 	}
 
-	efa_mr_cache_enable = 0;
-
+	free(domain->cache);
```

Comment:
Oh, God, this makes my eyes bleed.  I know this wasn't your fault originally, but we shouldn't write code that is hard to follow.  The primary control flow should be the happy case, with jumps to the error cases.  The happy case being buried in an if statement is awful.  Please flip line 322's logic to jump on error, and refactor so it is obvious this code is in the error path.

Reply:
It's not exactly an error path, since it will proceed without the cache...I'll try my best